### PR TITLE
Jenkinsfile and POM updates

### PR DESCRIPTION
**Pull request by jglick**
Pending [INFRA-1247](https://issues.jenkins-ci.org/browse/INFRA-1247).

@reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>2.21</version>
+    <version>2.30</version>
```
Comment (abayer) [inline pom.xml]: Shouldn't that be `2.31`?

Comment (jglick) [inline pom.xml]: Well, predated that POM. I did not bother to update it further at this time; surely that is not required.

**Comment by jglick** [comment]
For some reason the repository is now recognized, but the last commit was not. Maybe need to wait another day.

File: pom.xml (hunk 1 above)
Comment (abayer) [inline pom.xml]: Fair enough.

**Comment by jglick** [comment]
@reviewbybees done

(waiting for an actual CI build before merging)

**Pull request merged**